src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,38 +1,75 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import classNames from 'classnames';
+
+interface TodoModalProps {
+  todo: Todo;
+  onSetSelectedTodoId: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  onSetSelectedTodoId,
+  todo,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(todo.userId)
+      .then(userData => {
+        setUser(userData);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [todo.userId]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
-
-      {true ? (
-        <Loader />
+      {isLoading ? (
+        <Loader key="loader" />
       ) : (
         <div className="modal-card">
           <header className="modal-card-head">
             <div
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              Todo #{todo.id}
             </div>
-
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => onSetSelectedTodoId(0)}
+            />
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {todo.title}
             </p>
-
             <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+              <strong
+                className={classNames({
+                  'has-text-success': todo.completed,
+                  'has-text-danger': !todo.completed,
+                })}
+              >
+                {todo.completed ? 'Done' : 'Planned'}
+              </strong>
 
               {' by '}
-
-              <a href="mailto:Sincere@april.biz">Leanne Graham</a>
+              {user ? (
+                <a href={user.email}>{user.name}</a>
+              ) : (
+                <span>Loading user...</span>
```

Comment:
The conditional rendering for the user information is correct. Ensure that the `user` object has the necessary properties (`email` and `name`) and that they are correctly populated from the API response.